### PR TITLE
fix: jira worklog collector fetches only the 1st page

### LINKED ISSUE #7010: [Bug][Jira] Worklogs are not always imported completely
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

It seems that Worklogs are not always imported completely. In many cases, I see only the first page of Worklogs collected (20 items). I think that the expectations of this logic may not be met when the Worklogs are extracted from the Issue:

https://github.com/apache/incubator-devlake/blob/main/backend/plugins/jira/tasks/worklog_collector.go#L72

Additionally, I see cases where Epics have no Worklogs. I suspect that this may be due to the order of Tasks where Worklogs are extracted before Epics are collected. This is just a guess however.

### What do you expect to happen

All worklogs are imported from issues.

### How to reproduce

- Attempt to import an Issue with more than 20 Worklogs.

### Anything else

_No response_

### Version

v0.21.0-beta5

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by klesh**
### Summary
fix: jira worklog collector fetches only the 1st page

### Does this close any open issues?
Closes #7010

### Screenshots
all_worklog_collected
issue_worklog_total

### CODE REVIEW

File: backend/plugins/jira/models/migrationscripts/20240514_add_worklog_total_to_issue.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	// force full collection for all jira worklogs
+	return db.Exec("DELETE FROM _devlake_collector_latest_state WHERE raw_data_table = ?", "_raw_jira_api_worklogs")
+}
+
+func (*addWorklogToIssue) Version() uint64 {
+	return 20240514145131
+}
+
+func (*addWorklogToIssue) Name() string {
+	return "add worklog to _tool_jira_issues"
```
Comment (abeizn) [inline backend/plugins/jira/models/migrationscripts/20240514_add_worklog_total_to_issue.go:55]: worklog total

File: backend/plugins/jira/tasks/worklog_collector.go [hunk 2]
```diff
@@ -59,32 +60,15 @@ func CollectWorklogs(taskCtx plugin.SubTaskContext) errors.Error {
...
 	// filter out issue_ids that needed collection
 	clauses := []dal.Clause{
-		dal.Select("i.issue_id, i.updated AS update_time"),
+		dal.Select("i.issue_id AS issue_id, i.updated AS update_time"),
 		dal.From("_tool_jira_board_issues bi"),
 		dal.Join("LEFT JOIN _tool_jira_issues i ON (bi.connection_id = i.connection_id AND bi.issue_id = i.issue_id)"),
-		dal.Join("LEFT JOIN _tool_jira_worklogs wl ON (wl.connection_id = i.connection_id AND wl.issue_id = i.issue_id)"),
-		dal.Where("i.updated > i.created AND bi.connection_id = ?  AND bi.board_id = ?  ", data.Options.ConnectionId, data.Options.BoardId),
-		dal.Groupby("i.issue_id, i.updated"),
+		dal.Where("bi.connection_id=? and bi.board_id = ? AND i.std_type != ? and i.worklog_total > 20", data.Options.ConnectionId, data.Options.BoardId, "Epic"),
```
Comment (abeizn) [inline backend/plugins/jira/tasks/worklog_collector.go:66]: Will the first 20 items be collected? Is it collected in the issue_collector?

**Comment by abeizn** [comment]
e2e failed

File: backend/plugins/jira/tasks/worklog_collector.go (hunk 2 above)
Comment (klesh) [reply backend/plugins/jira/tasks/worklog_collector.go:66]: yes, it is, same as the changlog

**Review by abeizn — APPROVED** [review]

**Pull request merged**